pkg/apis/internal.acorn.io/v1/memory.go

```diff
@@ -0,0 +1,123 @@
+package v1
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"cuelang.org/go/cue/literal"
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+func ParseMemory(s []string) (Memory, error) {
+	result := Memory{}
+	for _, s := range s {
+		workload, memBytes, specific := strings.Cut(s, "=")
+
+		// If setting all, swap workload and memBytes
+		if !specific {
+			memBytes = workload
+			workload = ""
+		}
+
+		quantity, err := ParseQuantityString(memBytes)
+		if err != nil {
+			return Memory{}, err
+		}
+
+		result[workload] = &quantity
+	}
+	return result, nil
+}
+
+func ParseQuantityString(memory string) (int64, error) {
+	numInfo := literal.NumInfo{}
+	err := literal.ParseNum(memory, &numInfo)
+	if err != nil {
+		return -1, err
+	}
+
+	quantity, err := strconv.ParseInt(numInfo.String(), 10, 64)
+	if err != nil {
+		return -1, err
+	}
+
+	return quantity, nil
+}
+
+var (
+	ErrInvalidAcornMemory   = errors.New("invalid memory from Acornfile")
+	ErrInvalidSetMemory     = errors.New("invalid memory set by user")
+	ErrInvalidDefaultMemory = errors.New("invalid memory default")
+)
+
+// +k8s:deepcopy-gen=false
```

Comment:
I was iffy on this type before you added this comment. Now I think it would be valuable to figure out a way to not need this.

Reply:
Yeah I don't love this code. The only reason this struct exists is to pass off the invalid value so that API server's message is a bit friendlier.

Reply:
I'm happy to report that the `MemoryError` struct is gone!